src/modules/data/wrap_DataModule.cpp

```diff
@@ -302,25 +302,37 @@ int w_decode(lua_State *L)
 
 int w_hash(lua_State *L)
 {
-	const char *fstr = luaL_checkstring(L, 1);
+	ContainerType ctype = luax_checkcontainertype(L, 1);
```

Comment:
Could we keep the original two-parameter variant but mark it as deprecated (with love's deprecation APIs)?

Reply:
This has been resolved. The image shows the deprecated message, along with checking that it returns string type as it did in the previous version without the container type defined. ![image](https://github.com/love2d/love/assets/56016593/d0d9eb17-4fa9-4f5f-ace1-dd7994f0345d)